### ISSUE TITLE
What does the headless config option do?

### ISSUE THREAD

**Issue by stennie** [keystonejs/keystone#4809]
The `headless` config option has sparse documentation and no apparent test cases. It would be helpful to have more information on the intended use case(s).

From: https://keystonejs.com/documentation/configuration:

> You can disable the Admin UI by setting the `headless` option to `true`.
>
> This will allow you to use `keystone.start()` or `keystone.set('routes', aRouter.)` without Keystone creating route bindings for the Admin UI routes under `/keystone`.

### Expected behavior

The `headless` option should disable Keystone Admin UI routes but leave Keystone API routes available for use as a headless CMS.

### Actual/Current behavior

The `headless` option disables all keystone routes, including `/keystone/api`.

If this is the expected outcome, the documentation needs to be clearer. Otherwise this would be a bug.

### Environment

| Software         | Version
| ---------------- | -------
| Keystone         | 4.0.0

**Comment by Elemino** [comment]
Yeah! I'm getting the same thing. Would be awesome to have this working! Unless I'm doing something wrong..
